examples/pipelines/commoncrawl/components/image_url_deduplication/src/main.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+import dask.dataframe as dd
+
+from fondant.component import DaskTransformComponent
+
+logger = logging.getLogger(__name__)
+
+
+def deduplicate_dask_dataframe(dataframe: dd.DataFrame, column: str):
+    logger.info(f"Start exact deduplication of column {column}.")
+
+    num_partitions = dataframe.npartitions
+
+    # Sort the DataFrame on the column
+    dataframe = dataframe.sort_values(column)
+
+    # Set index to column
+    # dataframe = dataframe.set_index(column)
+
+    # Repartition on column
+    dataframe = dataframe.repartition(npartitions=num_partitions)
+
+    # Dedup partition wise on exact match
+    return dataframe.drop_duplicates(subset=[column], keep="last")
```

Comment:
Does this leverage the fact that the data is sorted? I think this is a global operation, while we sort first to make it a local operation.

Reply:
We also might want to implement the local operation ourselves so we can choose a strategy to keep a certain license. I see two options:

- Iterate through the rows while comparing each `url_surtkey` with the previous one, and then checking the license to decide which one to keep.
- Sort each partition on a second column as well, which should sort the licenses on strictness, so we can use `pandas.drop_duplicates` with the `keep` argument.

Reply:
Good point. I have implemented the second approach.